.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,64 @@
+name: integration-tests
+on:
+  push:
+    branches: [ "518-implement-semi-automatic-integration-tests" ]
+  pull_request:
+    branches: [ "518-implement-semi-automatic-integration-tests" ]
+
+env:
+  STORAGE_PATH: "/tmp"
+
+jobs:
+  deploy-and-test:
+    strategy:
+      matrix:
+        key: [8, 16]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checks-out repo
+        uses: actions/checkout@v2
+      - name: Install python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+      - name: Install deps
+        run: |
+          /usr/bin/pip3 install -r ./requirements.txt
+        working-directory: ./integration-tests
+      - name: Install compose
+        uses: KengoTODA/actions-setup-docker-compose@main
+        with:
+          version: '2.6.0'
+      - name: Get version data
+        run: |
+          echo "VERSION=$(cat bob-apps/Cargo.toml | grep ^version | head -1 | awk '{print $3}' | sed -e 's/"//g' | sed -e 's/-/./g')" >> $GITHUB_ENV
+          echo "VERSION_NO_FORMATTING=$(cat bob-apps/Cargo.toml | grep ^version | head -1 | awk '{print $3}' | sed -e 's/"//g')" >> $GITHUB_ENV
+      - name: Get tools
+        uses: dawidd6/action-download-artifact@v2
```

Comment:
There is a default action for downloading artifacts: `actions/download-artifact@v2`

Reply:
It seems that `actions/download-artifact@v2` does not support downloading artifacts from other workflows, so it is necessary to use this one instead.

Reply:
There is no need to download artifacts from other workflow. The proper way is to build all binaries within this workflow